src/server/zset_family.cc

```diff
@@ -737,6 +737,40 @@ OpResult<ScoredMap> OpUnion(EngineShard* shard, Transaction* t, string_view dest
   return result;
 }
 
+OpResult<ScoredMap> OpUnion(EngineShard* shard, Transaction* t, AggType agg_type,
+                            const vector<double>& weights) {
+  ArgSlice keys = t->ShardArgsInShard(shard->shard_id());
```

Comment:
It seems like this OpUnion is almost identical to the already present one if calling with an empty `dest` argument, except for calculating `windex`. Maybe we can just extend the already present one?